internal/app/handlers/handlers.go

```diff
@@ -18,20 +25,56 @@ type Handler struct {
 	storage repository.Repository
 }
 
-type RequestJSON struct {
-	URL string `json:"url"`
+func NewHTTPHandler(cfg *configs.Config, s repository.Repository) *Handler {
+	return &Handler{config: cfg, storage: s}
 }
 
-type ResponseJSON struct {
-	Result string `json:"result"`
+func extractUserID(r *http.Request) string {
+	ctx := r.Context().Value(middleware.UserCtx)
+	userID := ctx.(string)
```

Comment:
наверное можно в одну строку.  Но вообще кастуй типы с осторожностью, вдруг там не string. Откровенно говоря у меня ребята на проекте уже стреляли таким образом в ногу когда jwt токен обрабатывали) Хотя в текущем контексте можно сказать все ок, однако лучше подумать тогда над неймингом ctx
```suggestion
	userID, ok  := r.Context().Value(middleware.UserCtx).(string) 
```

Reply:
Пофиксил, надеюсь теперь ок)